openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/remote/ServerConnector.java

```diff
@@ -0,0 +1,249 @@
+				setIfIBM();
+			} catch (IOException ie) {
+				Message.logOut("Problem setting IBM extensions flag");
+				ie.printStackTrace();
+			} 
+		}
+	}
+
+	private void getSecureServerConnection(String user, String pw) {
+		// Define a JMX connector and connect to it securely using credentials
+		JMXConnector connector = null;
+		JMXServiceURL address = null;
+		Map<String, String[]> props = null;
+		try {
+			address = new JMXServiceURL(this.buildServerURL());
+		} catch (MalformedURLException mue) {
+			Message.logOut("One part of the JMXServiceURL is syntactically incorrect, " +
+					"or host is null and it is not possible to find the local host name, " + 
+					"or port is negative");
+			mue.printStackTrace();
+		} 
+
+		props = new HashMap<String, String[]>();
+		String[] credentials = new String[] { user, pw };
+		props.put("jmx.remote.credentials", credentials);             
+
+		// If we didn't connect keep try - the jvm being monitored might not be 
+		// ready to accept connections yet 
+		boolean connected = false;
+		int attempts = 0;
+		while (connected == false && attempts < 30) {        
+			try {    
+				connector = JMXConnectorFactory.connect(address, props);
+				connected = true;
+			} catch (IOException ie) {
+				try {           	  
+					Thread.sleep(5000);
+				} catch (InterruptedException e) {
+					Message.logOut("The sleeping profiler was interrupted");
+					e.printStackTrace();
+				}
+
+				Message.logOut("Monitored VM not ready at " + DateFormat.getDateTimeInstance().format(new Date()) +" (attempt " + attempts + ").");
+				Message.logOut("Wait 5 secs and trying again...");	
+
+				attempts++;           	
+				connected = false;
+
+				if (attempts == 30) {
+					Message.logOut("Failed to connect to Monitored VM after 30 attempts - giving up.  Connection Exception received is below:");	 
+					ie.printStackTrace();        	  	  
+				}
+			} 
+		}
+
+		try {
+			// Find out if we're running on an IBM JVM
+			if (connected == true) {	
+				Message.logOut("Connection established!");
+				this.mbs = connector.getMBeanServerConnection();
+				setIfIBM();
+			}
+		} catch (IOException ie) {
+			Message.logOut("Problem setting IBM extensions flag");
+			ie.printStackTrace();
+		}
+
+	}
+
+	// If we are running on an IBM JVM, we will want to renew our connection to make use 
+	// of the IBM-specific functionality
+	public MBeanServerConnection getNewConnection() {
+		if (secure) {
+			if (mbs == null) {
+				getSecureServerConnection(user,pw);
+			}
+			return mbs;
+		} else {
+			if (mbs == null) {
+				getServerConnection();
+			}
+			return mbs;
+		}
+	}
+
+	private void setIfIBM() throws IOException {
+		// Find out from the Runtime bean if we are running on an IBM JVM
+		RuntimeMXBean runtimeBean = ManagementFactory.newPlatformMXBeanProxy(mbs, 
+				ManagementFactory.RUNTIME_MXBEAN_NAME,
+				RuntimeMXBean.class);
+		String vendor = runtimeBean.getVmVendor();
+		if (vendor.contains("IBM")) {
```

Comment:
Is this true for OpenJ9, I thought the vendor name changed for that SDK, as did java -version output?  We should check.

Also, I guess if it is true (and you are testing against an IBM SDK, what does this check and information do for the test?